### PR TITLE
Extend product listing page with new fields

### PR DESCRIPTION
**Pull request by csordasmarton**
* Show the number of runs for every product.
* Show the latest date when a run in a product was updated.

extend-product-page

### CODE REVIEW

File: api/v6/products.thrift [hunk 1]
```diff
@@ -35,11 +35,13 @@ struct Product {
...
-  6: bool              accessible,        // Indicates whether the current user can access this product.
-  7: bool              administrating     // Indicates that the current user can administrate the product.
-  8: shared.DBStatus   databaseStatus     // Indicates the database backend status.
+  5: bool              connected,            // True only if database status is OK.
+                                             // !DEPRECATED FLAG databaseStatus is used to get the status of the database.
+  6: bool              accessible,           // Indicates whether the current user can access this product.
+  7: bool              administrating,       // Indicates that the current user can administrate the product.
+  8: shared.DBStatus   databaseStatus,       // Indicates the database backend status.
+  9: i64               runCount,             // Number of runs in the product.
+  10: string           latestUpdatedRunDate, // Latest date from the runs when the run in the product was updated.
```
Comment (whisperity) [inline api/v6/products.thrift]: Is it good that this is a `string`? Are we sending dates as strings elsewhere too?

File: www/scripts/productlist/ListOfProducts.js [hunk 2]
```diff
@@ -139,7 +139,9 @@ function (declare, domClass, domConstruct, ItemFileWriteStore, topic,
         { name : '&nbsp;', field : 'status', cellClasses : 'status', width : '20px', noresize : true },
         { name : '&nbsp;', field : 'icon', cellClasses : 'product-icon', width : '40px', noresize : true },
         { name : 'Name', field : 'name', cellClasses : 'product-name', width : '25%' },
-        { name : 'Description', field : 'description', styles : 'text-align: left;', width : '70%' }/*,
+        { name : 'Description', field : 'description', styles : 'text-align: left;', width : '70%' },
+        { name : 'Number of runs', field : 'runCount', styles : 'text-align: center;', width : '25%' },
+        { name : 'Date of latest run update', field : 'latestRunUpdateDate', styles : 'text-align: center;', width : '25%' }/*,
         { name : 'Last check date', field : 'date', styles : 'text-align: center;', width : '30%' },
```
Comment (whisperity) [inline www/scripts/productlist/ListOfProducts.js]: Remove this field, as the field before it implements the same functionality.

File: www/scripts/productlist/ListOfProducts.js [hunk 3]
```diff
@@ -139,7 +139,9 @@ function (declare, domClass, domConstruct, ItemFileWriteStore, topic,
         { name : '&nbsp;', field : 'status', cellClasses : 'status', width : '20px', noresize : true },
         { name : '&nbsp;', field : 'icon', cellClasses : 'product-icon', width : '40px', noresize : true },
         { name : 'Name', field : 'name', cellClasses : 'product-name', width : '25%' },
-        { name : 'Description', field : 'description', styles : 'text-align: left;', width : '70%' }/*,
+        { name : 'Description', field : 'description', styles : 'text-align: left;', width : '70%' },
+        { name : 'Number of runs', field : 'runCount', styles : 'text-align: center;', width : '25%' },
+        { name : 'Date of latest run update', field : 'latestRunUpdateDate', styles : 'text-align: center;', width : '25%' }/*,
```
Comment (whisperity) [inline www/scripts/productlist/ListOfProducts.js]: Is it good that we say `update` here? Perhaps there could be a less potentially misleading (and shorter) column?

`Latest store to product`

The fact that it is a "date" is trivial from looking at the possible values.

File: api/v6/products.thrift (hunk 1 above)
Comment (csordasmarton) [reply api/v6/products.thrift]: Yes, we are sending dates as strings in multiple places:

https://github.com/Ericsson/codechecker/blob/b9987109e6dac5aff4babf9bac47a34d3ad539e4/api/v6/report_server.thrift#L115
https://github.com/Ericsson/codechecker/blob/b9987109e6dac5aff4babf9bac47a34d3ad539e4/api/v6/report_server.thrift#L130
https://github.com/Ericsson/codechecker/blob/b9987109e6dac5aff4babf9bac47a34d3ad539e4/api/v6/report_server.thrift#L163-L164

**Pull request merged**